src/task.c

```diff
@@ -748,14 +802,28 @@ task_system_create(platform_heap_id    hid,
    // task initialization
    register_init_tid_hook();
 
+   // Treat this as a soft-error and configure the task system w/o
+   // background threads when one of these configs is 0.
+   bool use_bg_threads = ((num_bg_threads[TASK_TYPE_MEMTABLE] != 0)
+                          && (num_bg_threads[TASK_TYPE_NORMAL] != 0));
+
    ts->use_bg_threads = use_bg_threads;
    ts->heap_id        = hid;
    ts->scratch_size   = scratch_size;
    ts->init_tid       = INVALID_TID;
 
+   // Ensure that the main thread gets registered and init'ed first before
+   // any background threads are created. (Those may grab their own tids.).
+   task_run_thread_hooks(ts);
```

Comment:
Doesn't this break all the code we've already written where the main thread must call register_thread?

Reply:
Actually, there is no change in logic here. I'm just documenting that the main thread has to register itself first.

May be... I'm missing your question [?] ...

Reply:
Does the main thread need to call register_thread?  I thought that was taken care of automatically?  
https://github.com/vmware/splinterdb/blob/83de5e03722d61436ce247e406d87b8fa4913d85/include/splinterdb/splinterdb.h#L108-L109

Reply:
Yes, @rosenhouse  - the main thread does not have to call register_thread() itself. That happens, somewhat indirectly, when `task_system_create()` executes:

```
780 platform_status
781 task_system_create(platform_heap_id    hid,
[...]
802    // task initialization
803    register_init_tid_hook();
[...]
815    // Ensure that the main thread gets registered and init'ed first before
816    // any background threads are created. (Those may grab their own tids.).
817    task_run_thread_hooks(ts);                           <<<<<<<<<<
818    const threadid tid      = platform_get_tid();
```

Now, I'm not sure if your and Rob's questions are answered. And, if you are expecting / suggesting a rewording in some comments somewhere?

Reply:
Now that I see that this function was always called, I think you moved it to a better place.  LGTM.